### DISCUSSION TITLE
Watch newly registered field

### DISCUSSION THREAD

**Discussion by markivancho** [react-hook-form/react-hook-form#3717] [Q&A] [answered]
Steps
- go to CSB (https://codesandbox.io/s/react-hook-form-watch-v6-forked-7pted)
- open console
- click "Toggle age field"
- see watch() returned values without age field

Is there a possibility to have a field value returned by watch in cases like this?

Note: setting defaultValues in useForm is not an option, those field names are dynamic and conditional, their default value is controlled by Controller

**Comment by bluebill1049** [comment, marked as answer]
works for me? do I miss the steps?
<img width="682" alt="Screen Shot 2020-12-18 at 8 33 09 am" src="https://user-images.githubusercontent.com/10513364/102546378-a98d7d80-410b-11eb-9d93-dff77180ab77.png">

**Reply by markivancho** [reply]
Don't interact with anything after clicking Toggle
![image](https://user-images.githubusercontent.com/30623974/102591746-325fe580-411b-11eb-8843-977a006aa2c1.png)

**Reply by bluebill1049** [reply]
`watch` is not the same as state update, it only subscribes for the input change

**Reply by markivancho** [reply]
In this case, how can I get the value (that's controlled by Controller's `defaultValue`) of `age` field when it's mounted?

**Reply by bluebill1049** [reply]
put that in `defaultValues` instead of inline.

**Reply by markivancho** [reply]
and there is no other way than `deafultValues`?

**Reply by bluebill1049** [reply]
or supply a defaultValue: https://react-hook-form.com/api#watch
<img width="1000" alt="Screen Shot 2020-12-19 at 7 53 55 pm" src="https://user-images.githubusercontent.com/10513364/102685321-ef456580-4233-11eb-9a8f-47b6fe3870ea.png">

**Reply by markivancho** [reply]
alright, thanks for the answers!

**Reply by ghost** [reply]
> `watch` is not the same as state update, it only subscribes for the input change

the same for inputs registered with `methods.control`?

**Reply by bluebill1049** [reply]
what do you mean?

**Reply by ghost** [reply]
No, no, nothing. I was not getting the values of fields registered with `<Controller />`, but debugging it I found out that the cache from CRA somehow fuck the way React Hook Form works so I disabled it and I can get the values using `watch` now ( :

**Reply by bluebill1049** [reply]
nice. 👍

**Reply by uncvrd** [reply]
@markivancho sorry to revive an old thread, but I have the exact same scenario as you. I was curious if you had a code example of how you resolved this by receiving a value from `watch` just as it was mounted?
